nettests/nettests.go

```diff
@@ -67,252 +74,148 @@ func (c *Controller) SetNettestIndex(i, n int) {
+			c.Ctx.DB, reportID, exp.TestName, resultID, c.msmtPath, urlID,
+		)
 		if err != nil {
-			log.WithError(err).Error("Failed to create measurement")
-			return
-		}
-		c.msmts[idx] = msmt
-	})
-
-	nt.On("status.progress", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		perc := e.Value.Percentage
-		if c.ntCount > 0 {
-			perc = float64(c.ntIndex)/float64(c.ntCount) + perc/float64(c.ntCount)
+			return errors.Wrap(err, "failed to create measurement")
 		}
-		c.OnProgress(perc, e.Value.Message)
-	})
-
-	nt.On("status.update.*", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-	})
-
-	// XXX should these be made into permanent failures?
-	nt.On("failure.asn_lookup", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		log.Debugf("%v", e.Value)
-	})
-	nt.On("failure.cc_lookup", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		log.Debugf("%v", e.Value)
-	})
-	nt.On("failure.ip_lookup", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		log.Debugf("%v", e.Value)
-	})
-
-	nt.On("failure.resolver_lookup", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		log.Debugf("%v", e.Value)
-	})
-
-	nt.On("failure.report_close", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-		log.Debugf("%v", e.Value)
-	})
-
-	nt.On("failure.startup", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-
-		c.msmts[e.Value.Idx].Failed(c.Ctx.DB, e.Value.Failure)
-	})
-
-	nt.On("failure.measurement", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-
-		c.msmts[e.Value.Idx].Failed(c.Ctx.DB, e.Value.Failure)
-	})
-
-	nt.On("failure.measurement_submission", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-
-		failure := e.Value.Failure
-		c.msmts[e.Value.Idx].UploadFailed(c.Ctx.DB, failure)
-	})
+		c.msmts[idx64] = msmt
 
-	nt.On("status.measurement_submission", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-
-		// XXX maybe this should change once MK is aligned with the spec
-		if c.Ctx.Config.Sharing.UploadResults == true {
-			if err := c.msmts[e.Value.Idx].UploadSucceeded(c.Ctx.DB); err != nil {
-				log.WithError(err).Error("failed to mark msmt as uploaded")
+		measurement, err := exp.Measure(ctx, input)
+		if err != nil {
+			log.WithError(err).Debug(color.RedString("failure.measurement"))
+			if err := c.msmts[idx64].Failed(c.Ctx.DB, err.Error()); err != nil {
+				return errors.Wrap(err, "failed to mark measurement as failed")
 			}
+			continue
 		}
-	})
 
-	nt.On("status.measurement_done", func(e mk.Event) {
-		log.Debugf(color.RedString(e.Key))
-
-		if err := c.msmts[e.Value.Idx].Done(c.Ctx.DB); err != nil {
-			log.WithError(err).Error("failed to mark msmt as done")
+		// Make sure we share what the user wants us to share.
+		if c.Ctx.Config.Sharing.IncludeIP == false {
+			measurement.ProbeIP = model.DefaultProbeIP
+		}
+		if c.Ctx.Config.Sharing.IncludeASN == false {
+			measurement.ProbeASN = fmt.Sprintf("AS%d", model.DefaultProbeASN)
```

Comment:
Why is the `DefaultProbeASN` not a string with the `AS` prefix directly?

Reply:
Because the code model used in `ooni/probe-cli` used an `int` for that. I believe we can however define a `model.DefaultProbeASNString` constant in `probe-engine`, if you prefer that we have a canonical constant for that. Do you want me to do that?

Reply:
ping @hellais

Reply:
I guess you should use the type which is most comfortable. I would advise against defining another representation for it called `model.DefaultProbeASNString` as that would also be inconsistent with the naming of other variables.

If you are fine with doing the `fmt.Sprintf` calls then go for it. It was just something that made me think enough to write a comment while reviewing it.

I am fine with whatever solution you go for (even just leaving it as-is).

Reply:
ACK, thank you!